components/Panel/MenuPanel/MenuPanel.tsx

```diff
@@ -1,62 +1,66 @@
-import { useConnectWallet, useWallets } from "@web3-onboard/react";
-import { Button, getAccountDetails, handleDisconnectWallet, Nav, WalletIcon } from "components";
-import { useWalletContext } from "hooks";
+import { useConnectWallet } from "@web3-onboard/react";
+import { Button, Nav, WalletIcon } from "components";
+import { handleDisconnectWallet } from "helpers";
+import { useUserContext, useWalletContext } from "hooks";
 import styled from "styled-components";
 import { PanelFooter } from "../PanelFooter";
 import { PanelWrapper } from "../styles";
 
-const links = [
-  {
-    title: "Vote",
-    href: "/",
-  },
-  {
-    title: "Settled Disputes & Votes",
-    href: "/settled",
-  },
-  {
-    title: "Two Key Voting",
-    href: "/two-key",
-  },
-  {
-    title: "Optimistic Oracle",
-    href: "https://oracle.umaproject.org",
-  },
-  {
-    title: "Docs",
-    href: "https://docs.umaproject.org",
-  },
-];
-
 export function MenuPanel() {
-  const [{ wallet }, _connect, disconnect] = useConnectWallet();
+  const [_wallets, connect, disconnect] = useConnectWallet();
   const { setSigner, setProvider } = useWalletContext();
-  const connectedWallets = useWallets();
-  const { address } = getAccountDetails(connectedWallets);
+  const { address, connectedWallet, walletIcon } = useUserContext();
+
+  const links = [
+    {
+      title: "Vote",
+      href: "/",
+    },
+    {
+      title: "Settled Disputes & Votes",
+      href: "/settled",
+    },
+    {
+      title: "Wallet Settings",
+      href: "/wallet-settings",
+    },
+    {
+      title: "Optimistic Oracle",
+      href: "https://oracle.umaproject.org",
+    },
+    {
+      title: "Docs",
+      href: "https://docs.umaproject.org",
+    },
+  ];
+
+  const _links = connectedWallet ? links : links.filter((link) => link.href !== "/wallet-settings");
```

Comment:
We don't show the "wallet settings" link when there is no wallet connected to avoid confusion.